pkg/security/http/authz/and_authorizer.go

```diff
@@ -0,0 +1,76 @@
+package authz
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/Peripli/service-manager/pkg/log"
+	"github.com/Peripli/service-manager/pkg/web"
+
+	httpsec "github.com/Peripli/service-manager/pkg/security/http"
+)
+
+type andAuthorizer struct {
+	authorizers []httpsec.Authorizer
+}
+
+func NewAndAuthorizer(authorizers ...httpsec.Authorizer) httpsec.Authorizer {
+	return &andAuthorizer{
+		authorizers: authorizers,
+	}
+}
+
+// Authorize allows the request only if all nested authorizers allow it and denies even if one denies
+func (a *andAuthorizer) Authorize(request *web.Request) (httpsec.Decision, web.AccessLevel, error) {
+	ctx := request.Context()
+	logger := log.C(ctx)
+
+	errs := compositeError{}
+	abstain := false
+	denied := false
+
+	accessLevels := make([]web.AccessLevel, 0)
+	for _, authorizer := range a.authorizers {
+		decision, level, err := authorizer.Authorize(request)
+		if err != nil {
+			logger.WithError(err).Debug("AndAuthorizer: error during evaluate authorizer")
+			if decision == httpsec.Deny {
+				logger.Debug("AndAuthorizer: one authorizer denied, stop evaluating")
+				denied = true
+				errs = append(errs, err)
+				continue
+			}
+			return httpsec.Deny, web.NoAccess, err
+		}
+
+		if decision == httpsec.Deny {
+			denied = true
+		} else if decision == httpsec.Allow {
+			accessLevels = append(accessLevels, level)
+		} else {
+			abstain = true
+		}
+	}
+
+	if denied {
+		return httpsec.Deny, web.NoAccess, errs
+	} else if abstain {
```

Comment:
It is necessary because there is the third case in which it is allowed, but not explicitly abstained. That is why we have all the variables: `denied` and `abstain`